third-party/realdds/include/realdds/dds-stream-sensor-bridge.h

```diff
@@ -0,0 +1,110 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2023 Intel Corporation. All Rights Reserved.
+
+#pragma once
+
+#include <realdds/dds-stream-server.h>
+
+#include <map>
+#include <vector>
+#include <memory>
+#include <functional>
+
+
+namespace realdds {
+
+
+// Bridge between a device-server that's stream-based to a sensor-based paradigm: with the assumption
+// that every stream has an associated sensor, we need a mechanism that translates stream requests to
+// sensor requests.
+//
+// E.g., if a sensor 'Stereo Module' has three streams 'Depth', 'Infrared_1', 'Infrared_2', streaming
+// the first will have to start the sensor, then the 2nd will have to use the already-started one. Only
+// when all streams are closed will the sensor close.
+//
+// This is how things usually work in RealSense devices, so this class is useful for anything trying to
+// bridge a RealSense device, e.g. a tool like rs-dds-server.
+//
+// By default, we obviously do not know anything about RealSense and therefore do not actually start or
+// stop anything; instead we call notification functions that need implementing to customize behavior...
+//
+class dds_stream_sensor_bridge
+{
+    struct stream_bridge
+    {
+        std::shared_ptr< dds_stream_server > server;
+        std::shared_ptr< dds_stream_profile > profile;  // locked in when sensor starts
+        bool is_explicit = false;   // open() was called for the stream; it's not implicit via commit()
+        bool is_on = false;         // sensor has it open, either explicitly or implicitly; implies sensor.is_streaming
+        bool is_streaming = false;  // this stream has readers and we need to publish data on it; implies is_on
```

Comment:
If you have 2 flags that imply each other, then you can probably merge them to one

Reply:
This is changing, updating some time today...